Update KeyVault article for v10 (#4064)
umbraco/UmbracoDocs: Extending/Key-Vault/index-v10.md

---
versionFrom: 10.0.0
meta.Title: "Azure Key Vault"
meta.Description: "A guide configuring an Azure Key Vault"
---

# Configuring an Azure Key Vault

From a security perspective, it's always a good solution to store your application secrets (such as a connection string and others keys) in an Azure Key Vault. This article tells you how to configure your application so that it's ready to use a Key Vault

## Installing the package

Before you begin, you need to install the `Azure.Extensions.AspNetCore.Configuration.Secrets` and the `Azure.Identity` NuGet packages. There are two approaches to installing the packages:

1. Use your favorite IDE and open up the NuGet Package Manager to search and install the packages
1. Use the command line to install the package

> ## Installing through command line

Reviewer: Looks to me like a subheading to the "Installing the package" section? I could be wrong though. What do you think?

```suggestion
### Installing through command line
```

> Navigate to your project folder, which is the folder that contains your .csproj file. Now use the following 'dotnet add package' command to install the package:

Reviewer: ```suggestion
Navigate to your project folder, which is the folder that contains your `.csproj` file. Now use the following `dotnet add package` command to install the package:
```

```
dotnet add package Azure.Extensions.AspNetCore.Configuration.Secrets
dotnet add package Azure.Identity
```

## Configuration

> The next step is to add the Azure Key Vault endpoint to the 'appsettings.json' file. 

Reviewer: ```suggestion
The next step is to add the Azure Key Vault endpoint to the `appsettings.json` file. 
```

```json
{
  "AzureKeyVaultEndpoint": "https://{your-key-vault-name}.vault.azure.net",
}
```

> After adding the Key Vault endpoint you have to update the `CreateHostBuilder` method which you can find in the `Program.cs` class. 

Reviewer: I realize that it is a "class", but I think in the context here, it makes more sense to use "file", as it's a file you need to open and edit, right? 🤔 

```suggestion
After adding the Key Vault endpoint you have to update the `CreateHostBuilder` method which you can find in the `Program.cs` file. 
```

```csharp
public static IHostBuilder CreateHostBuilder(string[] args) =>
    Host.CreateDefaultBuilder(args)
        .ConfigureLogging(x => x.ClearProviders())
        .ConfigureAppConfiguration((context, config) =>
        {
            var settings = config.Build();
            var keyVaultEndpoint = settings[Constants.Azure.AzureKeyVaultEndpoint];
            if (!string.IsNullOrEmpty(keyVaultEndpoint) && Uri.TryCreate(keyVaultEndpoint, UriKind.Absolute, out var validUri))
            {
                config.AddAzureKeyVault(validUri, new DefaultAzureCredential());
            }
        })
        .ConfigureWebHostDefaults(webBuilder => webBuilder.UseStartup<Startup>());
```

## Authentication 

> There are several ways to access the Azure Key Vault. It is important that the user you are logging in with has access to the Key Vault. You can assign roles using the Azure Portal. 

Reviewer: ```suggestion
There are different ways to access the Azure Key Vault. It is important that the user you are logging in with has access to the Key Vault. You can assign roles using the Azure Portal. 
```
